File: comparative_tests/go_tests/basic_functions_tests/read_image.go
package basic_functions_tests

import (
	"fmt"
	"image"
	"image/jpeg"
	"os"
)

func ReadImage(filePath string) (img image.Image) {
	// Open the image file and check for errors
	file, err := os.Open(filePath)
	if err != nil {
		fmt.Println("Error opening file : ", err)
		return
	}
	// Close the file when the function returns
	defer func(file *os.File) {
		err := file.Close()
		if err != nil {
			fmt.Println("Error closing file:", err)
		}
	}(file)

	// Decode the image file and check for errors
	img, err = jpeg.Decode(file)
	if err != nil {
		fmt.Println("Error decoding file : ", err)
		return
	}

	return
}

File: comparative_tests/python_tests/basic_functions/read_image.py
from PIL import Image

def read_image(file_path):
    # Open the image file and check for errors
    try:
        img = Image.open(file_path)
    except IOError as e:
        print(f"Error opening file: {e}")
        return None

    # Decode the image file and check for errors
    try:
        img.load()  # Ensure the image data is read
    except IOError as e:
        print(f"Error decoding file: {e}")
        return None

    return img